### DISCUSSION TITLE
Question about tripping a generator

### DISCUSSION THREAD

**Discussion by xiaoheimai95** [CURENT/andes#498] [Q&A] [answered]
Dear Dr. Cui and his crew,
Hello! When I encountered the following problems in modeling, I would like to consult you:
When I impose a fault, then add a Toggle to trip `REGCV1 10` in order to control the rotor power angle in emergency control, I find that there is still a `REGCV1 10` curve when drawing the rotor power angle curve, does it mean that `REGCV1` failed to cut off? How to solve this problem? The following is a brief code:
```
case = andes.get_case("/Users/ieee39/ieee39_3_REGCV1.xlsx")
ss = andes.load(case, setup=False, no_ouput=True)
ss.add("Toggle", dict(model='RenGen', dev="REGCV1_10", t=5.0))
ss.setup()
ss.PQ.config.p2p = 1
ss.PQ.config.q2q = 1
ss.PQ.config.p2z = 0
ss.PQ.config.q2z = 0
ss.PQ.pq2z = 0
ss.PFlow.run()
ss.TDS.config.criteria = 0
ss.TDS.config.tf = 10
ss.TDS.run()
```
After the simulation, the pic is:
<img width="618" alt="截屏2023-12-05 20 25 14" src="https://github.com/CURENT/andes/assets/71546641/ac7b3e0b-b66e-408a-8ae8-cd1a9bd138a5">

**Comment by jinningwang** [comment]
I don't think so, a tripped generator still has value in the variables `Pe` and `omega`. You can check the output power, `Pe`, it should be zero after the event.

**Reply by xiaoheimai95** [reply]
Here is the pic of `Pe` and `omega`, the `Pe` of `REGCV1 10` is zero after the event, but the `omega` still has value.
<img width="603" alt="截屏2023-12-06 09 30 04" src="https://github.com/CURENT/andes/assets/71546641/ee30a419-c85e-4735-8328-a5427a2b0cd5">
<img width="614" alt="截屏2023-12-06 09 30 30" src="https://github.com/CURENT/andes/assets/71546641/32bba400-8ae0-4d2e-95b9-0ead8f873170">

**Reply by jinningwang** [reply]
Given the omega's equation in Documentation, I feel it's reasonable to have non-zero omega? In REGCV1, omega is an emulated variable rather than one has actual physical meaning.

**Reply by xiaoheimai95** [reply]
Does it mean that the omega and delta value of the tripped generator are just mathematically, I can just ignore them when consider further research?

**Reply by jinningwang** [reply, marked as answer]
It's reasonable to have non-zero omega values after being tripped for a REGCV1, they are "emulated" variables. Instead, omega and delta are "actual" variables for a SynGen.
I don't see any major trouble to skip these two non-zero values after being tripped.
